tests/packages/dynamic_metadata/plugin_project.toml

```diff
@@ -4,12 +4,17 @@ build-backend = "scikit_build_core.build"
 
 [project]
 name = "fancy"
-dynamic = ["readme", "version"]
+dynamic = ["readme", "version", "optional-dependencies"]
 
 [tool.scikit-build.metadata]
 version.provider = "scikit_build_core.metadata.setuptools_scm"
 readme.provider = "scikit_build_core.metadata.fancy_pypi_readme"
 
+[tool.scikit-build.metadata.optional-dependencies]
+provider = "scikit_build_core.metadata.template"
+needs = ["version"]
```

Comment:
If a metadata is changed from dynamic to static, should we display a warning, an error or just move on with the processing ?

And similarly, what should we do is a metadata does not exist ... should we error out ?

Reply:
>If a metadata is changed from dynamic to static, should we display a warning, an error or just move on with the processing ?

Don't think it's necessary. There is already a check that the field (`tool.scikit-build.metadata.<field>`) must be in `project.dynamic` and if a field is both static and in `project.dynamic` than it would already fail.

This is also probably where the terminology is affecting me. There is a difference between `settings` and `metadata` right, as in the fields in the `[project]` table and the `METADATA` fields that are tied to a specific `Metadata-Version`. I am mostly assuming we are only dealing with the former.

Reply:
Explicit needs is no longer part of this, it's automatic now. :)